packages/wrangler/src/deployment-bundle/source-maps.ts

```diff
@@ -0,0 +1,136 @@
+ * directory metadata.
+ */
+function loadSourceMap(
+	{ name, filePath }: CfModule,
+	sourceMapPath: string,
+	{ entryDirectory }: SourceMapMetadata
+): CfWorkerSourceMap[] {
+	if (filePath === undefined) {
+		return [];
+	}
+	const map = JSON.parse(
+		fs.readFileSync(sourceMapPath).toString()
+	) as SourceMapV3;
+	// Overwrite the file property of the source map to match the name of the
+	// main module in the multipart upload.
+	map.file = name;
+	// Remove the temporary directory prefix generated by Wrangler that appears
+	// in the source root path.
+	const sourceRootPath = path.dirname(path.join(entryDirectory, sourceMapPath));
+	map.sourceRoot = stripPrefix(sourceRootPath, map.sourceRoot);
+	map.sources = map.sources.map((source) => {
+		const originalPath = path.join(path.dirname(filePath), source);
+		// Remove the temporary build directory prefix generated by Wrangler
+		// that appears in the source path.
+		return stripPrefix(withTrailingSep(entryDirectory), originalPath);
+	});
+	return [
+		{
+			name: name + ".map",
+			content: JSON.stringify(map),
+		},
+	];
+}
+
+/**
+ * Find source maps by scanning module contents for special `//#
+ * sourceMappingURL=` comments pointing to source map files.
+ */
+function scanSourceMaps(modules: CfModule[]): CfWorkerSourceMap[] {
+	const maps: CfWorkerSourceMap[] = [];
+	for (const module of modules) {
+		const commentPrefix = "//# sourceMappingURL=";
+		const lastLine = module.content.toString().split("\n").pop();
+		if (
+			lastLine === undefined ||
+			!lastLine.startsWith(commentPrefix) ||
+			module.filePath === undefined
+		) {
+			continue;
+		}
+		// Assume the source map path in the comment is relative to the
+		// generated file it appears in.
+		const commentPath = stripPrefix(commentPrefix, lastLine).trim();
+		if (commentPath.startsWith("data:")) {
+			throw new Error(
+				`Invalid source map path in ${module.filePath}: expected file path but found data URL.`
+			);
+		}
+		// Convert source map path to an absolute path that we can read.
+		const wranglerPath = path.join(path.dirname(module.filePath), commentPath);
+		if (!fs.existsSync(wranglerPath)) {
+			throw new Error(
+				`Invalid source map path in ${module.filePath}: ${wranglerPath} does not exist.`
+			);
+		}
+		const map = JSON.parse(
+			fs.readFileSync(wranglerPath).toString()
+		) as SourceMapV3;
+		// Overwrite the file property of the sourcemap to match the name of the
+		// corresponding module in the multipart upload.
+		map.file = module.name;
+		map.sourceRoot = cleanPathPrefix(map.sourceRoot);
+		map.sources = map.sources.map(cleanPathPrefix);
+		maps.push({
+			name: module.name + ".map",
+			content: JSON.stringify(map),
+		});
+	}
+	return maps;
+}
+
+/**
+ * Removes leading "." and ".." segments from the given file path.
+ */
+function cleanPathPrefix(filePath: string): string {
+	return stripPrefix("..", stripPrefix("." + path.sep, filePath));
```

Comment:
Should this be:

```suggestion
	return stripPrefix(".." + path.sep, stripPrefix("." + path.sep, filePath));
```

Moreover it is not guaranteed that these paths will have the same path separator as the current OS. I.e. on Windows `path.sep` will be `\` but these paths here might well have `/`.

I would be lenient and actually handle both `/` and `\` in thes path.

Reply:
Similarly in the `withTrailingSep()` function below.